extractor/src/main/java/org/schabi/newpipe/extractor/services/media_ccc/extractors/MediaCCCStreamExtractor.java

```diff
@@ -100,64 +104,55 @@ public List<Image> getUploaderAvatars() {
-                    mediaFormat = MediaFormat.OGG;
-                } else {
-                    mediaFormat = null;
-                }
-
-                final AudioStream.Builder builder = new AudioStream.Builder()
-                        .setId(recording.getString("filename", ID_UNKNOWN))
-                        .setContent(recording.getString("recording_url"), true)
-                        .setMediaFormat(mediaFormat)
-                        .setAverageBitrate(UNKNOWN_BITRATE);
-
-                final String language = recording.getString("language");
-                // If the language contains a - symbol, this means that the stream has an audio
-                // track with multiple languages, so there is no specific language for this stream
-                // Don't set the audio language in this case
-                if (language != null && !language.contains("-")) {
-                    builder.setAudioLocale(LocaleCompat.forLanguageTag(language).orElseThrow(() ->
-                        new ParsingException(
-                                "Cannot convert this language to a locale: " + language)
-                    ));
-                }
-
-                // Not checking containsSimilarStream here, since MediaCCC does not provide enough
-                // information to decide whether two streams are similar. Hence that method would
-                // always return false, e.g. even for different language variations.
-                audioStreams.add(builder.build());
+        for (final MediaCCCRecording.Audio recording : recordings) {
+            // First we need to resolve the actual video data from the CDN
+            final MediaFormat mediaFormat;
+            if (recording.mimeType.endsWith("opus")) {
+                mediaFormat = MediaFormat.OPUS;
+            } else if (recording.mimeType.endsWith("mpeg")) {
+                mediaFormat = MediaFormat.MP3;
+            } else if (recording.mimeType.endsWith("ogg")) {
+                mediaFormat = MediaFormat.OGG;
+            } else {
+                mediaFormat = null;
             }
+            audioStreams.add(new AudioStream.Builder()
+                    .setId(recording.filename)
+                    .setContent(recording.url, true)
+                    .setMediaFormat(mediaFormat)
+                    .setAverageBitrate(UNKNOWN_BITRATE)
+                    .setAudioLocale(recording.language)
+                    .build());
         }
         return audioStreams;
     }
 
     @Override
     public List<VideoStream> getVideoStreams() throws ExtractionException {
-        final JsonArray recordings = data.getArray("recordings");
+
+        final List<MediaCCCRecording.Video> recordings = getRecordings().stream()
+                .flatMap(r ->
+                        r instanceof MediaCCCRecording.Video
+                                ? Stream.of((MediaCCCRecording.Video) r)
+                                : Stream.empty()
+                )
```

Comment:
(also in other places)
```suggestion
                .filter(MediaCCCRecording.Video::isInstance)
                .map(MediaCCCRecording.Video::cast)
```